### PR TITLE
Page logic finished

### PR DESCRIPTION
**Pull request by ChangweiXu**
Now pages can redirect to each other and communicate with Firebase.

### CODE REVIEW

**Comment by ChangweiXu** [comment]
@ryantse Ryan, this branch integrates TestDatabase and QuestionPages, so the previous pull request can be ignored.

File: src/main/resources/templates/partials/navbar.ftl [hunk 1]
```diff
@@ -7,16 +7,18 @@
...
+  <a href="#" class="btn btn-info my-2 my-sm-0 ml-2 justify-content-end"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABkAAAAZCAYAAADE6YVjAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAKySURBVEhLzZVdaI5hGMfftWkbLaJQZDnhxPLdZpyQHVIOZk05YEq02ebMRygHOKD5SFGGsNR8tHIg7UApaTTsxAnT8lEO1RQHXn7/y38vj/fZG2vFv37d9/W/rvu5n3f3/VzL/FfKZrO7v1nMv0K1U+MnHtoFmzw/A82RGIt40cl+4HpbIeLnsNjzHXAuEog1RcTtcJx5se10eYOHcBlewlm8mYx66DCUq46xFoZgA8yCu/AAeqGz4EYUXIGrfrMKuAif4BQsc9nIm6+DW/AZDurBMJH5U2hxab5I1sEQxVNsySv1NFXkJ3mq+QL4AJW20kVBB1xz+MdiTSno3DbbGl0UzYBhfk2ZrRDedGjHP8rYBHE+IyJeAa8cFhaFLdDhMESsP+Nb6IS9cBtewHyXhIj7IW5gnkjo+l2HAdAtyhUy1y97B8tthYi3QJ8O3Ja8nfARHsEFcvVOxW3RV9zIsIgxcdDEbVrgMCH8Z7DSYYhn6DOohv0wYPvHJp7micJjsMdhQvg3ocFhQvhV8HMTAvUitYrtUMOeRU4pp0PucZgTNSX4g7DQlmrL8dcy6s/fDf1ORVI/rxnOgw64zqlYCDrkrbZGNjgN3bZCxIdB1/gkbIPExciJRAN0OQwRz4PHoAfcgNcep7pEG08gHmScY2t0UVgJXyBxAVistrEK6slV2c4JL/qZNrOVLgrUk+7BPlt/Jdb1wBGH6aJAX/R9iLtPrFaxJJIpok5nU6uXU8x8GryB1VGQJpInoJdF6qa6gjoDtXS1/18PvgwO4enfgc7nDuijnW1v9P7FwmIK1N7VrtVNo5hRt0+doMLxGtCHOJc1Jd7wPehKt6mmoLzRLki0a+I+qPG8FX7vb3qRRodjE5tf4iEHGJcy6vo2OTV+4qEb2eCJYK5flf6h/RtlMt8BfFRJSv0q+7IAAAAASUVORK5CYII="></a>
```
Comment (ryantse) [inline src/main/resources/templates/partials/navbar.ftl]: This change reverts the usage of files within the img tag and goes back to using base64 encoded images.

Comment (ChangweiXu) [reply src/main/resources/templates/partials/navbar.ftl]: Resolved

File: src/main/resources/templates/question-page.ftl [hunk 2]
```diff
@@ -30,9 +29,9 @@
                         <a href="/question-list" float="right">View all the questions</a>
 		    			</div>
 		    			<!-- Here is the question body -->
-		    			<div class="vote-cell">
-    								<#include "./partials/vote_button.ftl" />
-    						</div>
+		    			<!-- <div class="vote-cell">
+								<#include "./partials/vote_button.ftl" />
```
Comment (ryantse) [inline src/main/resources/templates/question-page.ftl]: This <#include> is still interpreted when it's within a  block. This causes the rest of the page to not be rendered correctly.

**Pull request merged**